tests/0103-transactions.c

```diff
@@ -635,9 +635,9 @@ static void do_test_misuse_txn(void) {
                 error = rd_kafka_begin_transaction(p);
                 TEST_ASSERT(error, "Expected begin_transactions() to fail");
                 TEST_ASSERT(rd_kafka_error_code(error) ==
-                                RD_KAFKA_RESP_ERR__STATE,
+                                RD_KAFKA_RESP_ERR__CONFLICT,
```

Comment:
Is it possible to keep to same _STATE error here? Maybe some application or binding is expecting the previous code and not the new one. It's also similar to the exception Java throws: IllegalStateException

Reply:
Users are not supposed to treat specific error codes differently, but look at the error attributes (retriable, requires_txn_abort, etc), and the reason for using __CONFLICT is to make it clearer that it is primarily conflictuous. The __STATE error code is still returned for more statey errors, like calling commit_..() when not in a transaction and so forth.